src/append_only_data.rs

```diff
@@ -808,34 +909,62 @@ impl Data {
         }
     }
 
+    /// Returns the kind.
     pub fn kind(&self) -> Kind {
         self.address().kind()
     }
 
+    /// Returns the name.
     pub fn name(&self) -> &XorName {
         self.address().name()
     }
 
+    /// Returns the tag.
     pub fn tag(&self) -> u64 {
         self.address().tag()
     }
 
+    /// Returns `true` if published.
     pub fn is_pub(&self) -> bool {
         self.kind().is_pub()
     }
 
+    /// Returns `true` if unpublished.
     pub fn is_unpub(&self) -> bool {
         self.kind().is_unpub()
     }
 
+    /// Returns `true` if sequenced.
     pub fn is_seq(&self) -> bool {
         self.kind().is_seq()
     }
 
+    /// Returns `true` if unsequenced.
     pub fn is_unseq(&self) -> bool {
         self.kind().is_unseq()
     }
 
+    /// Checks permissions for given `action` for the provided user.
```

Comment:
Needs to be expanded to describe which error types we return here and in which cases.